### ISSUE TITLE
[BUG] Invalid local rotation

### ISSUE THREAD

**Issue by Geniok** [ezEngine/ezEngine#877]
**Bug description:**
When you select any object and enter degrees of rotation along the Y axis, the degrees of rotation along the X and Z axes also change.

**How to Reproduce:**
1. Open any project, for example "Sample Game".
2. Open the "Default" scene.
3. select the "Floor" geometry.
4. Set the Y-axis to 180.

**Screenshots:**
![2023-02-27_154137](https://user-images.githubusercontent.com/3386000/221566668-a6470f27-5500-46fe-899e-bc44d2d2e704.png)

**Comment by jankrassnigg** [comment]
This is an artifact from representing rotations as yaw, pitch and roll. It IS the correct rotation, you just don't see the values that you'd expect.

Internally the rotation is stored as a quaternion and thus there is only one representation of values (x,y,z,w which are not intuitive). For the UI we take the quaternion and convert it to YPR. However, there are IIRC 16 different possibilities, of which we have to pick one. When you change that value, the YPR representation is converted to quaternion, and then the result is converted back to YPR. At this point you'll notice that the YPR value can suddenly flip around wildly. Even though it still actually represents the same rotation that you wanted.

This is purely a UI issue. Though currently I don't see how to improve this.

**Comment by Geniok** [comment]
It seems logical to decide not to update the values that are already entered through ezQtPropertyEditorQuaternionWidget.

Otherwise, a double transformation is obtained:
Euler -> Quat , Quat -> Euler .

```
void ezQtPropertyEditorQuaternionWidget::InternalSetValue(const ngVariant& value)
{
	if (!m_bTemporaryCommand)
	{
                ezQtScopedBlockSignals b0(m_pWidget[0]);
		ezQtScopedBlockSignals b1(m_pWidget[1]);
		ezQtScopedBlockSignals b2(m_pWidget[2]);

		if (value.IsValid())
		{
			const ezQuat qRot = value.ConvertTo<ezQuat>();
			ezAngle x, y, z;
			qRot.GetAsEulerAngles(x, y, z);

			m_pWidget[0]->setValue(x.GetDegree());
			m_pWidget[1]->setValue(y.GetDegree());
			m_pWidget[2]->setValue(z.GetDegree());
		}
		else
		{
			m_pWidget[0]->setValueInvalid();
			m_pWidget[1]->setValueInvalid();
			m_pWidget[2]->setValueInvalid();
		}
	}
}
```

**Comment by jankrassnigg** [comment]
Yes, I've added a change that prevents doing this round trip while changing the value. I've also discovered a bug in the ToEuler conversion in general (that's what you get for copying code from Wikipedia I guess).
Note that the Euler value is never stored anywhere, so once the UI has to read the value, it always gets a quaternion and therefore has to convert it. Thus when you deselect an object and reselect it, the Euler values will still change. They just don't change anymore while you interact with the spinbox.
